modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-test/src/testIntegration/java/com/liferay/headless/commerce/admin/channel/resource/v1_0/test/ChannelResourceTest.java

```diff
@@ -290,6 +274,271 @@ private long _getCommerceChannelRelId() throws Exception {
 		return _commerceChannelRel.getCommerceChannelRelId();
 	}
 
+	private void _testPatchChannelByExternalReferenceCodeWithAccountExternalReferenceCode()
+		throws Exception {
+
+		Channel postChannel =
+			testPatchChannelByExternalReferenceCode_addChannel();
+
+		Channel randomPatchChannel = randomPatchChannel();
+
+		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
+			_user.getUserId(), 0, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null,
+			RandomTestUtil.randomString() + "@liferay.com", null,
+			RandomTestUtil.randomString(),
+			AccountConstants.ACCOUNT_ENTRY_TYPE_SUPPLIER,
+			WorkflowConstants.STATUS_APPROVED, _serviceContext);
+
+		randomPatchChannel.setAccountId(0L);
+		randomPatchChannel.setAccountExternalReferenceCode(
+			accountEntry.getExternalReferenceCode());
+
+		Channel patchChannel =
+			channelResource.patchChannelByExternalReferenceCode(
+				postChannel.getExternalReferenceCode(), randomPatchChannel);
+
+		randomPatchChannel.setAccountId(accountEntry.getAccountEntryId());
```

Comment:
why do you need this?

Reply:
because otherwise it compares all the object and the accountId is wrong because we forced it to be 0